pyspecdata/core.py

```diff
@@ -7255,25 +7209,26 @@ def rename(self,previous,new):
             self.fit_axis = new
         nddata.rename(self,previous,new)
         return self
-    def fit(self,set = None, set_to = None, force_analytical = False, silent = False):
+    def fit(self,set_what = None, set_to = None, force_analytical = False, silent = False):
         r'''actually run the fit'''
         if not silent: print("\n")
         if not silent: print(r'\resizebox*{!}{3in}{\begin{minipage}{\linewidth}')
-        if isinstance(set, dict):
-            set_to = list(set.values())
-            set = list(set.keys())
+        if isinstance(set_what, dict):
+            set_to = list(set_what.values())
+            set_what = list(set_what.keys())
         x = self.getaxis(self.fit_axis)
         if iscomplex(self.data.flatten()[0]):
-            if not silent: print(lsafen('Warning, taking only real part of fitting data!'))
+            if not silent: print((lsafen('Warning, taking only real part of fitting data!')))
```

Comment:
I would look for things in this function that refer to the variable `silent` and instead just do `logger.info(strm(....))` or `logger.debug(strm(...))`, as you think appropriate

Reply:
Ok, here I have replaced the print statement with logger statements, as that is what I took this to mean